packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -534,27 +546,40 @@ export function TransferPanel() {
   }
 
   const isTransferAllowed = useMemo(() => {
+    const isConnectedToTheWrongChain =
+      latestChain.current?.chain?.id !== latestNetworks.current.sourceChain.id
+
+    if (!arbTokenBridgeLoaded) {
+      return false
+    }
+    if (!latestEth.current) {
+      return false
+    }
```

Comment:
These conditions were in the while await Promise loop. 

Since the while loop is deleted, they are added here to prevent transfers.